api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -115,4 +131,37 @@ public static Tracer getTracer(String instrumentationName, String instrumentatio
   public static ContextPropagators getPropagators() {
     return get().getPropagators();
   }
+
+  // Use an inner class that checks and logs in its static initializer to have log-once behavior
+  // using initial classloader lock and no further runtime locks or atomics.
+  private static class SdkChecker {
+    static {
+      boolean hasSdk = false;
+      try {
+        Class.forName("io.opentelemetry.sdk.OpenTelemetrySdk");
+        hasSdk = true;
+      } catch (Throwable t) {
+        // Ignore
```

Comment:
might be marginally clearer if we just returned immediately in this case

Reply:
It's a static initializer so can't return

Reply:
you are so right. can static initializers call static methods that could return? Just curious...doesn't really matter that much.

Reply:
It can - but would just add some indirection so inclined to leave it like this.